File: sdk/node/src/types/groups.ts
export interface CreateGroupRequest {
  name: string;
  description?: string;
  external_id?: string;
}

export interface Group {
  id: string;
  organization_id?: string;
  name: string;
  description?: string;
  external_id?: string;
  member_count?: number;
  created_at?: string;
  updated_at?: string;
}

export interface UpdateGroupRequest {
  name?: string;
  description?: string;
}

export interface AddGroupMemberRequest {
  user_id: string;
  role?: string;
}

export interface GroupMember {
  id: string;
  user_id: string;
  group_id: string;
  role?: string;
  email?: string;
  full_name?: string;
  joined_at?: string;
}

export interface UpdateGroupMemberRequest {
  role: string;
}

export interface GroupRole {
  role_id: string;
  group_id: string;
  created_at?: string;
}

export interface CreateInvitationRequest {
  email: string;
  role_id?: string;
  metadata?: Record<string, any>;
  expires_in_days?: number;
}

export interface InvitationResponse {
  id: string;
  tenant_id: string;
  email: string;
  role_id?: string;
  status: string;
  invited_by?: string;
  expires_at?: string;
  created_at?: string;
}

export interface AcceptInvitationRequest {
  token: string;
  password: string;
  full_name: string;
  metadata?: Record<string, any>;
}

File: sdk/python/coreauth/types/groups.py
from __future__ import annotations

from typing import Any, Optional

from pydantic import BaseModel


class CreateGroupRequest(BaseModel):
    name: str
    description: Optional[str] = None
    external_id: Optional[str] = None


class Group(BaseModel):
    id: Optional[str] = None
    tenant_id: Optional[str] = None
    organization_id: Optional[str] = None
    name: Optional[str] = None
    description: Optional[str] = None
    external_id: Optional[str] = None
    member_count: Optional[int] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


class UpdateGroupRequest(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class AddGroupMemberRequest(BaseModel):
    user_id: str
    role: Optional[str] = None


class GroupMember(BaseModel):
    id: Optional[str] = None
    user_id: Optional[str] = None
    group_id: Optional[str] = None
    role: Optional[str] = None
    email: Optional[str] = None
    full_name: Optional[str] = None
    joined_at: Optional[str] = None


class UpdateGroupMemberRequest(BaseModel):
    role: str


class GroupRole(BaseModel):
    role_id: Optional[str] = None
    group_id: Optional[str] = None
    created_at: Optional[str] = None


class AssignGroupRoleRequest(BaseModel):
    role_id: str


class CreateInvitationRequest(BaseModel):
    email: str
    role_id: Optional[str] = None
    metadata: Optional[dict[str, Any]] = None
    expires_in_days: Optional[int] = None


class InvitationResponse(BaseModel):
    id: Optional[str] = None
    tenant_id: Optional[str] = None
    email: Optional[str] = None
    role_id: Optional[str] = None
    status: Optional[str] = None
    invited_by: Optional[str] = None
    expires_at: Optional[str] = None
    created_at: Optional[str] = None


class AcceptInvitationRequest(BaseModel):
    token: str
    password: str
    full_name: str
    metadata: Optional[dict[str, Any]] = None
